src/internal/streamstore/streamstore.go

```diff
@@ -0,0 +1,194 @@
+// streamstore implements helpers to store large
+// data streams in a repository
+package streamstore
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+
+	"github.com/pkg/errors"
+
+	"github.com/alcionai/corso/src/internal/data"
+	"github.com/alcionai/corso/src/internal/kopia"
+	"github.com/alcionai/corso/src/internal/stats"
+	"github.com/alcionai/corso/src/pkg/backup/details"
+	"github.com/alcionai/corso/src/pkg/path"
+)
+
+type streamStore struct {
+	kw       *kopia.Wrapper
+	resource string
+	tenant   string
+	service  path.ServiceType
+}
+
+func New(
+	kw *kopia.Wrapper,
+	tenant string,
+	service path.ServiceType,
+) *streamStore {
+	return &streamStore{kw: kw, resource: "metadata", tenant: tenant, service: service}
+}
+
+const detailsItemName = "details"
+
+// WriteBackupDetails persists a `details.Details`
+// object in the stream store
+func (ss *streamStore) WriteBackupDetails(
+	ctx context.Context,
+	backupDetails *details.Details,
+) (string, error) {
+	// construct the path of the container for the `details` item
+	p, err := path.Builder{}.
+		ToServiceCategoryMetadataPath(
+			ss.tenant,
+			ss.resource,
+			ss.service,
+			path.DetailsCategory,
+			false,
+		)
+	if err != nil {
+		return "", err
+	}
+
+	// TODO: We could use an io.Pipe here to avoid a double copy but that
+	// makes error handling a bit complicated
+	dbytes, err := json.Marshal(backupDetails)
+	if err != nil {
+		return "", errors.Wrap(err, "marshalling backup details")
+	}
+
+	dc := &streamCollection{
+		folderPath: p,
+		item: &streamItem{
+			name: detailsItemName,
+			data: dbytes,
+		},
+	}
+
+	backupStats, _, err := ss.kw.BackupCollections(ctx, nil, []data.Collection{dc}, ss.service)
```

Comment:
Per the discussion in the other PR about querying for specific manifests, this could embed a unique Tag for looking up (or filtering against) details snapshots.  That PR modified this func to pass in a specific backupID.  A more generic resolution to fit both cases would be to pass in a map of `Tags`.

Reply:
Going back to the other discussion, this suggestion is unnecessary at this time.  We can handle it in the future, when needed.